tests/networkpolicy_test.go

```diff
@@ -1,173 +1,264 @@
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: serverVMI.Namespace,
+						Name:      "deny-by-label",
+					},
+					Spec: networkv1.NetworkPolicySpec{
+						PodSelector: metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								"type": "test",
+							},
 						},
+						Ingress: []networkv1.NetworkPolicyIngressRule{},
 					},
-					Ingress: []v1network.NetworkPolicyIngressRule{},
-				},
-			}
-			var err error
-			policy, err = virtClient.NetworkingV1().NetworkPolicies(vmia.Namespace).Create(networkpolicy)
-			Expect(err).ToNot(HaveOccurred())
-		})
+				}
+				createNetworkPolicy(policy)
+			})
 
-		AfterEach(func() {
-			waitForNetworkPolicyDeletion(policy)
-		})
+			AfterEach(func() {
+				waitForNetworkPolicyDeletion(policy)
+			})
 
-		It("[test_id:1515] should be failed to reach vmia from vmic", func() {
-			By("Connect vmia from vmic")
-			assertPingFail(vmic, vmia)
-		})
+			It("[test_id:1515] should be failed to reach serverVMI from clientVMIAlternativeNamespace", func() {
+				By("Connect serverVMI from clientVMIAlternativeNamespace")
+				assertPingFail(clientVMIAlternativeNamespace, serverVMI)
+			})
 
-		It("[test_id:1516] should be failed to reach vmia from vmib", func() {
-			By("Connect vmia from vmib")
-			assertPingFail(vmib, vmia)
-		})
+			It("[test_id:1516] should be failed to reach serverVMI from clientVMI", func() {
+				By("Connect serverVMI from clientVMI")
+				assertPingFail(clientVMI, serverVMI)
+			})
 
-		It("[test_id:1517] should be successful to reach vmib from vmic", func() {
-			By("Connect vmib from vmic")
-			assertPingSucceed(vmic, vmib)
+			It("[test_id:1517] should be successful to reach clientVMI from clientVMIAlternativeNamespace", func() {
+				By("Connect clientVMI from clientVMIAlternativeNamespace")
+				assertPingSucceed(clientVMIAlternativeNamespace, clientVMI)
+			})
+
+		})
+		Context("and an allow all http ports policy is populated", func() {
+			var policy *networkv1.NetworkPolicy
+			BeforeEach(func() {
+				port80 := intstr.FromInt(80)
+				port81 := intstr.FromInt(81)
+				tcp := corev1.ProtocolTCP
+				policy = &networkv1.NetworkPolicy{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: serverVMI.Namespace,
+						Name:      "allow-all-http-ports",
+					},
+					Spec: networkv1.NetworkPolicySpec{
+						PodSelector: metav1.LabelSelector{},
+						Ingress: []networkv1.NetworkPolicyIngressRule{
+							{
+								Ports: []networkv1.NetworkPolicyPort{
+									{Port: &port80, Protocol: &tcp},
+									{Port: &port81, Protocol: &tcp},
+								},
+							},
+						},
+					},
+				}
+				createNetworkPolicy(policy)
+			})
+			AfterEach(func() {
+				waitForNetworkPolicyDeletion(policy)
+			})
+			It("[test_id:2774] should allow http traffic for ports 80 and 81 from clientVMI to serverVMI", func() {
+				PingHTTPEventually(clientVMI, serverVMI, 80).Should(Succeed(), "should allow HTTP traffic from clientVMI go serverVMI with port 80")
```

Comment:
I think that you should check that from the VMI on the other namespace, the ping fails.

Reply:
Done

Reply:
We are not testing k8s and we don't want to add more test time to already long CI.